dbms/src/Storages/tests/gtest_SplitTokenExtractor.cpp

```diff
@@ -0,0 +1,174 @@
+        {
+            "Multiple tokens separated by whitespace",
+            "\nabc 123\tXYZ\r",
+            {
+                "abc", "123", "XYZ"
+            }
+        },
+        {
+            "Multiple tokens separated by non-printable chars",
+            BINARY_STRING("\0abc\1" "123\2XYZ\4"),
+            {
+                "abc", "123", "XYZ"
+            }
+        },
+        {
+            "ASCII table is split into numeric, upper case and lower case letters",
+
+            BINARY_STRING("\x00\x01\x02\x03\x04\x05\x06\x07\x08\t\n\x0b\x0c\r\x0e\x0f\x10\x11\x12\x13\x14\x15\x16"
+            "\x17\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f !\"#$%&\'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNO"
+            "PQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~\x7f\x80\x81\x82\x83\x84\x85\x86\x87"
+            "\x88\x89\x8a\x8b\x8c\x8d\x8e\x8f\x90\x91\x92\x93\x94\x95\x96\x97\x98\x99\x9a\x9b\x9c"
+            "\x9d\x9e\x9f\xa0\xa1\xa2\xa3\xa4\xa5\xa6\xa7\xa8\xa9\xaa\xab\xac\xad\xae\xaf\xb0\xb1"
+            "\xb2\xb3\xb4\xb5\xb6\xb7\xb8\xb9\xba\xbb\xbc\xbd\xbe\xbf\xc0\xc1\xc2\xc3\xc4\xc5\xc6"
+            "\xc7\xc8\xc9\xca\xcb\xcc\xcd\xce\xcf\xd0\xd1\xd2\xd3\xd4\xd5\xd6\xd7\xd8\xd9\xda\xdb"
+            "\xdc\xdd\xde\xdf\xe0\xe1\xe2\xe3\xe4\xe5\xe6\xe7\xe8\xe9\xea\xeb\xec\xed\xee\xef\xf0"
+            "\xf1\xf2\xf3\xf4\xf5\xf6\xf7\xf8\xf9\xfa\xfb\xfc\xfd\xfe\xff"),
+            {
+                "0123456789", "ABCDEFGHIJKLMNOPQRSTUVWXYZ", "abcdefghijklmnopqrstuvwxyz",
+                "\x80\x81\x82\x83\x84\x85\x86\x87"
+                "\x88\x89\x8a\x8b\x8c\x8d\x8e\x8f\x90\x91\x92\x93\x94\x95\x96\x97\x98\x99\x9a\x9b\x9c"
+                "\x9d\x9e\x9f\xa0\xa1\xa2\xa3\xa4\xa5\xa6\xa7\xa8\xa9\xaa\xab\xac\xad\xae\xaf\xb0\xb1"
+                "\xb2\xb3\xb4\xb5\xb6\xb7\xb8\xb9\xba\xbb\xbc\xbd\xbe\xbf\xc0\xc1\xc2\xc3\xc4\xc5\xc6"
+                "\xc7\xc8\xc9\xca\xcb\xcc\xcd\xce\xcf\xd0\xd1\xd2\xd3\xd4\xd5\xd6\xd7\xd8\xd9\xda\xdb"
+                "\xdc\xdd\xde\xdf\xe0\xe1\xe2\xe3\xe4\xe5\xe6\xe7\xe8\xe9\xea\xeb\xec\xed\xee\xef\xf0"
+                "\xf1\xf2\xf3\xf4\xf5\xf6\xf7\xf8\xf9\xfa\xfb\xfc\xfd\xfe\xff"
+            }
+        }
+    })
+);
+
+
+INSTANTIATE_TEST_SUITE_P(SIMD_Cases,
+    SplitTokenExtractorTest,
+    ::testing::ValuesIn(std::initializer_list<SplitTokenExtractorTestCase>{
+        {
+            "First 16 bytes are empty, then a shor token",
+            "                abcdef",
+            {"abcdef"}
+        },
+        {
+            "Token crosses bounday of 16-byte chunk",
+            "            abcdef",
+            {"abcdef"}
+        },
+        {
+            "Token ends at the end of 16-byte chunk",
+            "          abcdef",
+            {"abcdef"}
+        },
+        {
+            "Token crosses bondaries of multiple 16-byte chunks",
+            "abcdefghijklmnopqrstuvwxyz",
+            {"abcdefghijklmnopqrstuvwxyz"}
+        },
+    })
+);
```

Comment:
Also, please add some padding tests, it can easily be done with PaddedPOD array with garbage padding set and then resized

Reply:
There is "implicit garbage" padding, but good point anyway, let's make sure that malicious padding does not break anything.

Reply:
done